src/core/Wallet.ts

```diff
@@ -18,16 +19,18 @@ export class Wallet<P extends IProvider> implements IWallet {
   readonly docs: string;
   readonly networks: Network[];
   readonly provider: P | null = null;
+  readonly hardware?: boolean;
   account: Account | null = null;
 
-  constructor({ id, origin, name, icon, docs, networks, provider }: WalletOptions<P>) {
+  constructor({ id, origin, name, icon, docs, networks, provider, hardware }: WalletOptions<P>) {
     this.id = id;
     this.origin = origin;
     this.name = name;
     this.icon = icon;
     this.docs = docs;
     this.networks = networks;
     this.provider = provider;
+    this.hardware = hardware;
```

Comment:
avoid of using flags

Reply:
Okay, let me explain the main idea behind the current implementation. Wallet and WalletConnector know nothing about wallet types cause we adopt everything to a single interface by using metadata + some core logic. Instead of adding type flag I've modified schema and added new parameter `label`. So instead of adding additional `if/else` or `switch/case` statements now you can use this new metadata parameter